File: ulohy/17-vlajecka/reseni/main.cpp
#include <iostream>

void printFlag(int poleLength, int flagWidth, int heightAboveGround) {
    int flagHeight = flagWidth * 2 - 1;

    if (flagHeight + heightAboveGround > poleLength) {
        std::cout << "invalid" << std::endl;
        return;
    }

    for (int i = 0; i < poleLength - flagHeight - heightAboveGround; i++) {
        std::cout << '|' << std::endl;
    }

    for (int i = 1; i <= flagWidth; i++) {
        for (int j = 1; j <= i; j++) {
            std::cout << '#';
        }
        std::cout << std::endl;
    }

    for (int i = flagWidth - 1; i >= 1; i--) {
        for (int j = 1; j <= i; j++) {
            std::cout << '#';
        }
        std::cout << std::endl;
    }

    for (int i = 0; i < heightAboveGround; i++) {
        std::cout << '|' << std::endl;
    }
}

int main() {
    int p, f, h;
    std::cin >> p >> f >> h;

    printFlag(p, f, h);

    return 0;
}

File: ulohy/17-vlajecka/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let p, f, h;

rl.on("line", (line) => {
    if (p === undefined) {
        p = parseInt(line);
        return;
    }
    else if (f === undefined) {
        f = parseInt(line);
        return;
    }
    h = parseInt(line);

    printFlag(p, f, h);

    rl.close();
});

function printFlag(poleLength, flagWidth, heightAboveGround) {
    const flagHeight = flagWidth * 2 - 1;

    if (flagHeight + heightAboveGround > poleLength) {
        console.log("invalid");
        return;
    }

    for (let i = 0; i < poleLength - flagHeight - heightAboveGround; i++) {
        console.log("|");
    }

    for (let i = 1; i <= flagWidth; i++) {
        for (let j = 1; j <= i; j++) {
            process.stdout.write("#");
        }
        process.stdout.write("\n");
    }

    for (let i = flagWidth - 1; i >= 1; i--) {
        for (let j = 1; j <= i; j++) {
            process.stdout.write("#");
        }
        process.stdout.write("\n");
    }

    for (let i = 0; i < heightAboveGround; i++) {
        console.log("|");
    }
}
